pandas/io/sas/sasreader.py

```diff
@@ -128,6 +128,7 @@ def read_sas(
     if iterator or chunksize:
         return reader
```

Comment:
The problem is potentialy here also, e.g. if the user forgets to wrap his code in a try/finally. 

Could it be an idea to add ``__enter__`` & ``__exit__``  to ReaderBase and rearrange to use ``with`` blocks, both here and below?

Reply:
What the user does with the Reader object is not pandas' concern ;)
The important thing is not to leave any created objects in an inconsistent state.